asyncscope.md

```diff
@@ -550,6 +543,45 @@ While `start_detached` is a customization point object, its correspondent, `asyn
 At this point, we do not believe that customizing the operations of `async_scope` will be needed by the users.
 Therefore, the paper does not propose any customization point objects.
 
+Q & A
+=====
+
+## Q why does async_scope terminate in the destructor instead of blocking like jthread?
+
+One author's perspective:
+
+`jthread` blocking in the destructor is bad for composition. 
+`jthread` and `thread` should `terminate()` if the destructor runs before the thread exits. 
+
+Imagine `make_shared<jthread>(..)`. 
+Where will the destructor run? 
+In what context will the destructor run? 
+
+We can require users to know whether the destructor blocks for every type, and require users to carefully control the lifetime of all those objects - with the only indication of failure being a deadlock. Or we can teach that destructors will not block and indicate lifetime failures with `terminate()`. 
+
+software is less likely to ship with crashes and more likely to be fixed when there are crashes. Deadlocks result in users forcefully terminating the app and forced terminations are rarely reported to the developer as a bug and even if reported, tend to have no debug data (stacks, dumps, etc..). If there is a lifetime bug that you want fixed - it had better crash.
+
+Blocking must be explicit (exiting a sync scope is implicit - and shared_ptr makes it even more scary). 
+Blocking must be grepable. 
+Blocking must be rare. 
+Blocking is like reinterpret_cast<> - the name should be long and scary. 
+`join()` is a terrible design - this is why async_scope has `on_empty()` instead. 
+Every asychronous lifetime must be managed with non-blocking primitives and only `sync_wait()` is used to block.
```

Comment:
I would also add here:
`sync_wait()` should as well be avoided as much as possible; blocking a thread to wait for some computation to complete almost always has performance costs.

Ideally we should have ony one `sync_wait()` in `main()`. Other valid uses for `sync_wait()` is for slideware.

"Blocking must be rare" seems to be an understatement :)